File: backend/src/main/java/com/foodya/foodya_backend/user/dto/UserProfileResponse.java
package com.foodya.foodya_backend. user.dto;

import java.time.LocalDateTime;
import java. util.UUID;

import com.foodya.foodya_backend.user.model.Role;
import com.foodya.foodya_backend.user.model.User;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserProfileResponse {
    private UUID id;
    private String username;
    private String email;
    private String fullName;
    private String phoneNumber;

    @Builder.Default
    private Role role = Role.CUSTOMER;

    private Boolean isActive;
    private Boolean isEmailVerified;
    private Boolean isPhoneNumberVerified;

    // Profile
    private String profileImageUrl;

    // Timestamps
    private LocalDateTime lastLoginAt;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    /**
     * Convert User entity to UserProfileResponse DTO
     */
    public static UserProfileResponse fromEntity(User user) {
        return UserProfileResponse.builder()
                .id(user.getId())
                .username(user.getUsername())
                .email(user. getEmail())
                .fullName(user.getFullName())
                .phoneNumber(user.getPhoneNumber())
                .role(user.getRole())
                .isActive(user.getIsActive())
                .isEmailVerified(user.getIsEmailVerified())
                .isPhoneNumberVerified(user.getIsPhoneNumberVerified())
                .profileImageUrl(user.getProfileImageUrl())
                .lastLoginAt(user.getLastLoginAt())
                .createdAt(user.getCreatedAt())
                .updatedAt(user.getUpdatedAt())
                .build();
    }
}

File: mobile/app/src/main/java/com/example/foodya/data/model/UserProfileResponse.kt
package com.example.foodya.data.model

import com.example.foodya.domain.model.User
import com.google.gson.annotations.SerializedName

data class UserProfileResponse(
    @SerializedName("id")
    val id: String,
    
    @SerializedName("username")
    val username: String,
    
    @SerializedName("email")
    val email: String,
    
    @SerializedName("fullName")
    val fullName: String,
    
    @SerializedName("phoneNumber")
    val phoneNumber: String?,
    
    @SerializedName("role")
    val role: String,
    
    @SerializedName("isActive")
    val isActive: Boolean,
    
    @SerializedName("isEmailVerified")
    val isEmailVerified: Boolean,
    
    @SerializedName("isPhoneNumberVerified")
    val isPhoneNumberVerified: Boolean? = null,
    
    @SerializedName("profileImageUrl")
    val profileImageUrl: String? = null,
    
    @SerializedName("lastLoginAt")
    val lastLoginAt: String?,
    
    @SerializedName("createdAt")
    val createdAt: String,
    
    @SerializedName("updatedAt")
    val updatedAt: String
) {
    fun toDomain(): User {
        return User(
            id = id,
            username = username,
            email = email,
            fullName = fullName,
            phoneNumber = phoneNumber ?: "",
            role = role,
            isActive = isActive,
            isEmailVerified = isEmailVerified,
            lastLoginAt = lastLoginAt ?: "",
            createdAt = createdAt,
            updatedAt = updatedAt
        )
    }
}
